plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapManagerImpl.java

```diff
@@ -243,4 +263,22 @@ public boolean isLdapEnabled() {
             closeContext(context);
         }
     }
+
+    @Override
+    public LinkDomainToLdapResponse linkDomainToLdap(Long domainId, String type, String name, short accountType) {
+        Validate.notNull(type, "type cannot be null. It should either be GROUP or OU");
+        Validate.notNull(domainId, "domainId cannot be null.");
+        Validate.notEmpty(name, "GROUP or OU name cannot be empty");
```

Comment:
If multiple CS domains map to same ldap group, then when the user is authn. against ldap and created in CS (lazy creation), under which domain does it go?